### ISSUE TITLE
Create a Krew Plugin to allow users to interact with Armada via kubectl

### ISSUE THREAD

**Issue by kannon92** [armadaproject/armada#2120]
Add armadactl to Krew, generate Krew manifest and make it installable as a kubectl plugin.

**Comment by 120EE0980** [comment]
Helloo..I want to contribute to this project idea as part of gsoc2023. I am new to kubernetes. @kannon92 can you provide me any resources where I can get started.

**Comment by kannon92** [comment]
I suggest trying to create an example kubectl plugin.

https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/

And after that it’s apply that knowledge to making armadactrl a kubectl plugin

**Comment by octonawish-akcodes** [comment]
Hi, I am interested in this project under gsoc 2023, @kannon92 can you provide some references

**Comment by kannon92** [comment]
> Also is there any slack link

https://github.com/armadaproject/armada/issues/2156

feel free to take that issue to make our slack more obvious and please join!

**Comment by ratul-07** [comment]
Hello, I want to be a part of this project under gsoc 2023, I know basics about docker and kubernetes, @kannon92  could you please point me to other valuable resources and is there any slack channel?

**Comment by kannon92** [comment]
> Hello, I want to be a part of this project under gsoc 2023, I know basics about docker and kubernetes, @kannon92 could you please point me to other valuable resources and is there any slack channel?

https://github.com/armadaproject/armada/blob/master/CONTRIBUTING.md

When I start a new project I like to read the docs.  See if I can build it and look for any gaps in their docs.

**Comment by Dinuka-Thathsara** [comment]
hello everyone,

I am a trainee software engineer in wso2 and a computer science and engineering undergraduate who works on docker, Kubernetes, and go technologies mostly and would like to start contributing to the Armada project as well! So nice to meet with the community! I have also found some interesting projects in gsoc 2023.
Additionally, since I am new to Armada Project, I would really appreciate it if you can point out some guidance or some good first issues. Thank you in advance!

**Comment by kannon92** [comment]
We are very interested in everyone's willingness to contribute.  

https://developers.google.com/open-source/gsoc/timeline 

I suggest getting familiar with our project.

I've created some other issues for people to play around with.  We are excited to have this much interest but I ask that you don't take this project unless you are officially allowed to do so via GSOC application.

**Comment by sarthaksarthak9** [comment]
. I am looking forward to learn and have knowledge of Kubernetes, docker.  I want to be a part of this project . Additionally, since I am new to Armada Project, I would really appreciate it if you can point out some guidance or some good first issues.

**Comment by Rahul-Kumar-prog** [comment]
> We are very interested in everyone's willingness to contribute.
> 
> https://developers.google.com/open-source/gsoc/timeline
> 
> I suggest getting familiar with our project.
> 
> I've created some other issues for people to play around with. We are excited to have this much interest but I ask that you don't take this project unless you are officially allowed to do so via GSOC application.

Hey @kannon92 can you tell us which issues you open to play around with to get familiar with the armada ?

**Comment by iyear** [comment]
Hello! I'm also very interested in this project. I learned about armada from CNCF's GSoC idea list, which is a great job scheduler.

It just so happens that the technology stack of the project fits me well. I've developed many CLI applications using Golang, and most of them have received praise from users. The experience of reading the kubectl source code will also help me learn krew quickly.

I'm tring to learn about armada and looking into the krew. Could you let me know about any materials or resources that I should learn to get a better understanding of the project and its requirements? Thanks!

**Comment by kannon92** [comment]
Please look at issues we tagged good-first-issues.  I also suggest making sure you can follow our docs and get a dev setup.

GSOC information is here: https://github.com/cncf/mentoring/discussions/848

**Comment by ashutosh887** [comment]
@kannon92  I would like to contribute to this Org. for GSOC
Please consider

**Comment by elijah0kello** [comment]
> I suggest trying to create an example kubectl plugin.
> 
> https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/
> 
> And after that it’s apply that knowledge to making armadactrl a kubectl plugin

Thanks for this @kannon92

**Comment by AryanSharma9917** [comment]
@elijah0kello sir I would like to work on this one what I can do is provides a "submit" subcommand that allows users to submit a batch job to a specified Kubernetes cluster with various configuration options, such as the number of tasks to run and input files and parameters for the job.

**Comment by cubxxw** [comment]
How many people do you need for this function? One person？

**Comment by kannon92** [comment]
> 

Yes. GSOC does not allow multiple people working on this.  For google summer of code, you have to write up a proposal for your implementation plan. CNCF chooses the application and that person implements this ticket.

**Comment by octonawish-akcodes** [comment]
CNCF chooses the right candidate?

**Comment by ShivangShandilya** [comment]
Greetings @kannon92, I'm intrigued with the Armada project. I'll start contributing right away but is there any stuff that I need to have specific knowledge of or something to get me a good grasp on how Armada works?

**Comment by ShivangShandilya** [comment]
> I suggest trying to create an example kubectl plugin.
> 
> https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/
> 
> And after that it’s apply that knowledge to making armadactrl a kubectl plugin

Ohh this helps a lot. Thanks Kevin

**Comment by VasuDevrani** [comment]
Hey @kannon92,
Is this issue still open and unresolved? If yes, I would like to work on this
Thanks

**Comment by kannon92** [comment]
This is a google summer of code ticket so please don’t take this one.

**Comment by VasuDevrani** [comment]
Okay, that's fine. I'll look for another

**Comment by kannon92** [comment]
This could be a good one.

https://github.com/armadaproject/armada/issues/2255
